src/app/my-rewards/backers/components/Table/MobileSortModal.tsx

```diff
@@ -0,0 +1,122 @@
+import { Modal } from '@/components/Modal/Modal'
+import { Button } from '@/components/Button'
+import { Paragraph } from '@/components/Typography'
+import { SelectableItem } from '@/components/SelectableItem'
+import { ColumnId, SORT_OPTIONS } from './BackerRewardsTable.config'
+import { FC, useState, useEffect } from 'react'
+import { TrashIcon } from '@/components/Icons'
+import { SORT_DIRECTION_ASC, SORT_DIRECTION_DESC } from '@/shared/context/TableContext/constants'
+import { SortDirection } from '@/shared/context/TableContext/types'
+
+interface MobileSortModalProps {
+  isOpen: boolean
+  currentSort: ColumnId | null
+  currentSortDirection: SortDirection | null
+  onClose: () => void
+  onApply: (sort: ColumnId | null, sortDirection: SortDirection | null) => void
+  onReset: () => void
+}
+
+export const MobileSortModal: FC<MobileSortModalProps> = ({
+  isOpen,
+  currentSort,
+  currentSortDirection,
+  onClose,
+  onApply,
+  onReset,
+}) => {
+  const [tempSort, setTempSort] = useState<ColumnId | null>(currentSort)
+  const [tempSortDirection, setTempSortDirection] = useState<SortDirection | null>(currentSortDirection)
+
+  const resetTempState = () => {
+    setTempSort(currentSort)
+    setTempSortDirection(currentSortDirection)
+  }
+
+  useEffect(() => {
+    if (isOpen) {
+      resetTempState()
+    }
+  }, [isOpen, currentSort, currentSortDirection]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  if (!isOpen) return null
+
+  const handleClose = () => {
+    resetTempState()
+    onClose()
+  }
+
+  const handleApply = () => {
+    onApply(tempSort, tempSortDirection)
+  }
+
+  const handleReset = () => {
+    setTempSort(null)
+    setTempSortDirection(null)
+    onReset()
+  }
+
+  const sortRadioOptions = SORT_OPTIONS.map(option => ({
+    label: option.label,
+    value: option.id,
+  }))
+
+  const sortDirectionOptions = [
```

Comment:
extract to module scope so that it doesn't get recreated on every render

```suggestion
  const SORT_DIRECTION_OPTIONS = [
```